openmc/data/thermal.py

```diff
@@ -670,12 +670,41 @@ def from_ace(cls, ace_or_filename, name=None):
                 mu_i = []
                 for j in range(n_energy_out[i]):
                     mu = ace.xss[idx + 4:idx + 4 + n_mu]
+                    # The equiprobable angles produced by NJOY are not always
+                    # sorted. This is problematic when the smearing algorithm
```

Comment:
I never realized this! Do you have an example of a particular library/nuclide/reaction/distribution for which this happens?

Reply:
So this is actually a common occurrence as well (though not systematically the case as with the CDFs). In ENFB/B-VIII.0, I have seen warnings from PapillonNDL about this for important evaluations like H-H2O (all temperatures), D-D2O (all temperatures), O-D2O (all temperatures), and several other materials as well. I don't think it happens for every single incident/outgoing energy pair, but it certainly happens at least once in each of those evaluations. PapillonNDL stops reading the ACE file after the first error, so its hard for me to say with certainty how prevalent it is. When bench marking PapillonNDL, I don't believe I ever noticed a statistical difference with OpenMC due to my sorting of the angles, but this is still less than ideal, as smearing assumes that they are sorted. If you want, I have programs which directly sample the distributions, either from PapillonNDL or OpenMC, and then a script which gets the p-value of the energy and angle distributions to compare.

Another issue I noticed with thee ENDF/B-VII.1 ACE files is that the scattering angles where sometimes larger than 1, or less than -1 (this happens in the ligher water TSL for sure). This was a bug in not-so-old versions of NJOY, but has since been fixed. I added a check for this to PapillonNDL, and because of that, I simply can't use the older ENDF/B-VII.0 ACE files that were distributed with MCNP for example. Currently, I don't see a check for this in the OpenMC processing. There isn't really a good solution for this problem, other than re-processing the evaluation with the newer version of NJOY. Do you want me to add a check for this @paulromano ? It might cause people problems if they want to generate HDF5 files from these older ACE files, but maybe this is better than letting them use physically invalid data ?

Reply:
I should specify, when I mention ENDF/B-VIII.0 here, these are the ACE files made from lib80x, distributed by LANL.

Reply:
Re: the check for values outside of [-1,1], perhaps the best thing to do would be to add a warning so that at least we're letting the user know they may be dealing with bad data. That doesn't necessarily need to be done as part of this PR though.